### PR TITLE
[DD-330]: Promote Share Button

### PR DESCRIPTION
**Pull request by tshen815**
[Ticket](https://circleci.atlassian.net/browse/DD-330)
[Preview](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/promote-share-btn-preview/?force-all)

NOTE: copying a section will not work in preview link due to `navigator.clipboard` needing a secure origin (HTTPS or localhost). You can get around this by going to `chrome://flags` and enabling `Insecure origins treated as secure` and adding `http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com` into the field. Remember to disable after! 

# Changes
- Remove experiment logic gating experiment to treatment
- extract share button functionality from `main.js` to separate `services/sectionShareButton.js` file
- initialize service in `app.js` 

# Description
We are making the experiment “Share Section button” permanent in prod due to positive experiment results: 

- averaging 170 hovers and 21 clicks, out of 424 daily views (avg) 
- 40% of views will hover
-  12% of user will click from hover

# Reasons
Original PR #5693 

# Amplitude Events

- `docs-share-button-click`

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4335,11 +4335,6 @@ highlightjs-badge@0.1.9:
   resolved "https://registry.yarnpkg.com/highlightjs-badge/-/highlightjs-badge-0.1.9.tgz#39cec0ff9d55d4ca02ea04ab31e376331f66d9b1"
   integrity sha512-uWA5OvsV3zTnzlXWv12AymBwWwH7VjzndiX/ts2O4NXnM6ezpbkjMKfB6WtYF84syNltoOZgMrluPvwI4utrUw==
 
-highlightjs-line-numbers.js@^2.8.0:
-  version "2.8.0"
-  resolved "https://registry.yarnpkg.com/highlightjs-line-numbers.js/-/highlightjs-line-numbers.js-2.8.0.tgz#479ea8cff0c31fadc1578a66fa03e38b801f9ca6"
-  integrity sha512-TEf1gw0c8mb8nan0QwliqS7obT4cpUd9hzsGzsZLweteNnWea/VIqy5/aQqsa5wnz9lnvmtAkS1ZtDTjB/goYQ==
-
```
Comment (rsylvian) [inline yarn.lock:4342]: Why is this in your diff?

File: src/js/services/sectionShareButton.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+              options: {
+                offset: [0, 8],
+              },
+            },
+          ],
+        });
+        window.AnalyticsClient.trackAction('docs-share-button-hover', {
+          page: location.pathname,
+          success: true,
+        });
```
Comment (rsylvian) [inline src/js/services/sectionShareButton.js:42]: I think we can remove that event now. I don't thing it gives us much insight now that we are out of the experimentation phase

**Review by rsylvian — APPROVED** [review]

**Review by rsylvian — CHANGES_REQUESTED** [review]
Getting some errors when clicking on this header in http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/promote-share-btn-preview/2.0/hello-world/
Screen Shot 2022-01-21 at 6 43 18 PM

Screen Shot 2022-01-21 at 6 43 14 PM

(I enabled the Chrome flag btw)

File: yarn.lock (hunk 1 above)
Comment (tshen815) [reply yarn.lock:4342]: It popped up after i yarn installed to use the newest yarn dev. I could do a git checkout and revert it?

**Comment by tshen815** [comment]
> Getting some errors when clicking on this header in http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/promote-share-btn-preview/2.0/hello-world/ Screen Shot 2022-01-21 at 6 43 18 PM
> 
> Screen Shot 2022-01-21 at 6 43 14 PM
> 
> (I enabled the Chrome flag btw)

Strange, it seems to be working for me well. Which url did you use for the chrome flag? Would love to see if other people are having the same issue.

https://user-images.githubusercontent.com/57234605/150624089-e25081f1-165c-4342-aa01-193c75575a7c.mp4

File: src/js/services/sectionShareButton.js (hunk 2 above)
Comment (tshen815) [reply src/js/services/sectionShareButton.js:42]: Good call. I was on the fence about the hover one too but I can remove it 👍

File: src/js/services/index.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import OptimizelyClient from './optimizely';
 import * as rum from './rum';
 import * as highlightjsBadge from './highlightJSBadge';
 import * as progressbar from './progressbar';
+import * as sectionShareButton from './sectionShareButton';
```
Comment (chantelle11) [inline src/js/services/index.js:8]: great idea to do this!

File: src/js/services/sectionShareButton.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  $('article h1, article h2, article h3, article h4, article h5, article h6')
+    .filter('[id]')
+    .each(function () {
+      var isMainTitle = $(this).prop('nodeName') === 'H1';
+      $(this).append(
+        (isMainTitle ? ' <a href="#' : '<a href="#' + $(this).attr('id')) +
+          '"><i class="fa fa-link"></i></a>',
+      );
+      if (isMainTitle) {
+        $(this).find('i').toggle();
```
Comment (chantelle11) [inline src/js/services/sectionShareButton.js:19]: what is 'i' ?

Comment (chantelle11) [reply src/js/services/sectionShareButton.js:19]: Oh I think its finding the icon, maybe we can use an ID instead?

**Review by rsylvian — APPROVED** [review]
LGTM! Tested it locally and it worked... not sure what happened on the preview with my browser ¯\_(ツ)_/¯

**Pull request merged**